index.js

```diff
@@ -258,15 +296,25 @@ async function run() {
 
       exportCredentials({accessKeyId, secretAccessKey, sessionToken});
     }
-
-    // Regardless of whether any source credentials were provided as inputs,
-    // validate that the SDK can actually pick up credentials.  This validates
-    // cases where this action is on a self-hosted runner that doesn't have credentials
-    // configured correctly, and cases where the user intended to provide input
-    // credentials but the secrets inputs resolved to empty strings.
-    await validateCredentials(accessKeyId);
-
-    const sourceAccountId = await exportAccountId(maskAccountId, region);
+    
+    // Attempt to load credentials from the GitHub OIDC provider.
+    // If a user provides an IAM Role Arn and DOESN'T provide an Access Key Id
+    // The only way to assume the role is via GitHub's OIDC provider.
+    let sourceAccountId;
+    let webIdentityToken;
+    if(useGitHubOIDCProvider()) {
+      webIdentityToken = await getWebIdentityToken(maskAccountId, region);
```

Comment:
Do we need to send maskAccountId and region to getWebIdentityToken for pulling OIDC credentials?

Reply:
fixed